application.ts

```diff
@@ -11,7 +12,7 @@ import { serve, Server } from "./deps.ts";
  * constructing an instance of `Application`.
  */
 export class Application<S extends object = { [key: string]: any }> {
-  private _middleware: Middleware<S, Context<S>>[] = [];
+  private _middleware: Middleware<any, Context<any>>[] = [];
```

Comment:
Maybe `unknown` us better here?

Reply:
unknown would got a `unknown does not satisfy the constraint "object"` compiling error, but 
I think it's OK to omit type parameters here.